src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -4547,7 +4547,40 @@ public static void Main(string[] args)
                 );
         }
 
-        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/16559")]
+        [Fact, WorkItem(16559, "https://github.com/dotnet/roslyn/issues/16559")]
+        public void CasePatternVariableUsedInCaseExpression()
```

Comment:
Please test SemanticModel behavior as well, we should have all the helpers for that.